docs/extensibility.md

```diff
@@ -1,564 +1,537 @@
-3. Implement your protocol and provide a full end to end sample.
-4. Create a pull request, updating the minor version of akri. See [contributing](./contributing.md#versioning) to learn more about our versioning strategy.
-
-For a protocol to be considered fully implemented the following must be included in the PR. Note how the Nessie protocol above only has completed the first 3 requirements. 
-1. A new DiscoveryHandler implementation in the Akri Agent
-1. An update to the Configuration CRD to include the new `ProtocolHandler`
-1. A sample protocol broker for the new resource
-1. A sample Configuration that uses the new protocol in the form of a Helm template and values
-1. (Optional) A sample end application that utilizes the services exposed by the Configuration
-1. Dockerfile[s] for broker [and sample app] and associated update to the [makefile](../build/akri-containers.mk)
-1. Github workflow[s] for broker [and sample app] to build containers and push to Akri container repository
-1. Documentation on how to use the new sample Configuration, like the [udev Configuration document](./udev-configuration.md)
+# Extensibility: HTTP-based Devices
+
+While Akri has several [currently supported discovery protocols](./roadmap.md#currently-supported-protocols) and sample brokers and applications to go with them, the protocol you want to use to discover resources may not be implemented yet. This walks you through all the development steps needed to implement a new protocol and sample broker. It will also cover the steps to get your protocol and broker[s] added to Akri, should you wish to contribute them back.
+
+To add a new protocol implementation, three things are needed:
+
+1. Add a new DiscoveryHandler implementation in Akri Agent
+1. Update Configuration CRD to include the new DiscoveryHandler implementation
+1. Create a (protocol) Broker for the new capability
+
+To demonstrate how new protocols can be added, we will create a protocol to discover HTTP-based devices that publish random sensor data. An implementation of these devices and a discovery protocol is described in this [README](./samples/apps/http-apps/README.md).
+
+See [Extensibility: HTTP protocol](https://github.com/deislabs/akri/issues/85)
+
+For reference, we have created a [http-extensibility](https://github.com/deislabs/akri/tree/http-extensibility) with the implementation defined below.  For convenience, you can [compare the http-extensibility branch with main here](https://github.com/deislabs/akri/compare/http-extensibility).
+
+Any Docker-compatible container registry will work (dockerhub, Github Container Registry, Azure Container Registry, etc).
+
+For this sample, we are using the [GitHub Container Registry](https://github.blog/2020-09-01-introducing-github-container-registry/). You can follow the [getting started guide here to enable it for yourself](https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry).
+
+## Agent
```

Comment:
i think having some introduction to why Agent chagnes are required ... something that links back to the "To add a new protocol implementaiton, three things are needed" list.  something like:

## New DiscoveryHandler implementation
If the resource you are interested in defining is not accessible through the [included protocols](./roadmap.md#currently-supported-protocols), then you will need to create a DiscoveryHandler for your new protocol.  Here, we will create a discovery handler in order to discover http resources.

Reply:
in a more generalized sense, since this is attempting to explain how to extend akri, maybe mix in some of the original extensibility.md's "this is why we are changing this code" bits of text.

Reply:
As, I see.

I'm almost out-of-puff rewriting docs.

I'm going to walk my dog.

I'll rewrite this tomorrow.

Reply:
totally understandable!  if you aren't against it, i can take a crack at it try to weave the commentary in.

Reply:
Thanks for being understanding... I feel it's better for me to be honest... Totally (!) not against it ;-)

Reply:
ok ... i'll see what i can do :)

Reply:
@DazWilkin, to shorten the document, i tried to combine the device and discovery mock servers ... can you take a look at the doc changes and verify that it makes sense?

Reply:
You bet! Thanks very much for doing this. I'll review first thing tomorrow.